CefSharp.WinForms/Handler/LifeSpanHandler.cs

```diff
@@ -0,0 +1,160 @@
+                if (!control.IsDisposed && control.IsHandleCreated)
+                {
+                    //We need to invoke in a sync fashion so our IBrowser object is still in scope
+                    //Calling in an async fashion leads to the IBrowser being disposed before we
+                    //can access it.
+                    control.InvokeSyncOnUiThreadIfRequired(new Action(() =>
+                    {
+                        onPopupDestroyed?.Invoke(control, browser);
+                    }));
+                }
+            }
+
+            //No WM_CLOSE message will be sent, manually handle closing
+            return true;
+        }
+
+        public LifeSpanHandler OnPopupDestroyed(Action<Control, IBrowser> onPopupDestroyed)
+        {
+            this.onPopupDestroyed = onPopupDestroyed;
+
+            return this;
+        }
+
+        /// <inheritdoc/>
+        protected override void OnAfterCreated(IWebBrowser chromiumWebBrowser, IBrowser browser)
+        {
+            if (browser.IsPopup)
+            {
+                var windowHandle = browser.GetHost().GetWindowHandle();
+
+                //WinForms will kindly lookup the child control from it's handle
+                //If no parentControl then likely it's a native popup created by CEF
+                //(Devtools by default will open as a popup, at this point the Url hasn't been set, so 
+                // we're going with this assumption as it fits the use case currently)
+                var control = Control.FromChildHandle(windowHandle) as ChromiumHostControl;
+
+                //On WinForms parent control so we'll treat this as a native popup and do nothing
+                if (control != null && !control.IsDisposed)
+                {
+                    control.BrowserHwnd = windowHandle;
+
+                    control.InvokeOnUiThreadIfRequired(() =>
+                    {
+                        var interceptor = new ParentFormMessageInterceptor(control);
+                        interceptor.Moving += (sender, args) =>
+                        {
+                            if (!browser.IsDisposed)
+                            {
+                                browser?.GetHost()?.NotifyMoveOrResizeStarted();
+                            }
+                        };
+
+                        popupParentFormMessageInterceptors.Add(browser.Identifier, interceptor);
+                    });                   
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override void OnBeforeClose(IWebBrowser chromiumWebBrowser, IBrowser browser)
+        {
+            if (!browser.IsDisposed && browser.IsPopup)
+            {
+                ParentFormMessageInterceptor interceptor;
+
+                if (popupParentFormMessageInterceptors.TryGetValue(browser.Identifier, out interceptor))
+                {
+                    popupParentFormMessageInterceptors[browser.Identifier] = null;
```

Comment:
Why not remove it? Or do you want to throw on `Add` if an identifier gets reused?

```suggestion
                    popupParentFormMessageInterceptors.Remove(browser.Identifier);
```